src/main/java/org/mockbukkit/mockbukkit/inventory/InventoryViewMock.java

```diff
@@ -145,14 +145,52 @@ public void setCursor(@Nullable ItemStack item)
 	@Override
 	public void setItem(int slot, @Nullable ItemStack item)
 	{
-		// TODO Auto-generated method stub
+		int topSize = getTopInventory().getSize();
+		if (slot < 0)
+		{
+			// TODO : ???
+			throw new UnimplementedOperationException();
+		}
+
+		if (slot < topSize)
+		{
+			getTopInventory().setItem(slot, item);
+			return;
+		}
+
+		slot -= topSize;
+		if (slot < getBottomInventory().getSize())
+		{
+			getBottomInventory().setItem(slot, item);
+			return;
+		}
+
+		// TODO: ???
 		throw new UnimplementedOperationException();
 	}
 
 	@Override
 	public @Nullable ItemStack getItem(int slot)
 	{
-		// TODO Auto-generated method stub
+		int topSize = getTopInventory().getSize();
+		if (slot < 0)
+		{
+			// TODO : ???
+			throw new UnimplementedOperationException();
```

Comment:
Same here

Reply:
Just tested: it returns null

Reply:
Alright, let's be more consistent on the mockbukkit side and just throw index out of bounds exception.